tests/Aspire.Hosting.Tests/DistributedApplicationTests.cs

```diff
@@ -744,7 +757,7 @@ public async Task ProxylessContainerWithoutPortThrows()
     {
         using var builder = TestDistributedApplicationBuilder.Create();
 
-        var redis = builder.AddRedis("redis").WithEndpoint("tcp", endpoint =>
+        var redis = builder.AddContainer("dummyRedis", "redis").WithEndpoint("tcp", endpoint =>
```

Comment:
When change from `AddRedis` to `AddContainer` here?

Reply:
Because `AddRedis` sets the TargetPort

Reply:
So it succeeds and I cannot test the error condition